libosmscout-client/src/osmscoutclient/MapManager.cpp

```diff
@@ -43,6 +43,11 @@ CancelableFuture<bool> MapManager::LookupDatabases()
     std::vector<std::filesystem::path> databaseFsDirectories;
 
     for (const auto &lookupDir:databaseLookupDirs){
+      if (!std::filesystem::exists(lookupDir) || !std::filesystem::is_directory(lookupDir)) {
```

Comment:
and what about symlinks?

Reply:
I have edited the code to resolve symlinks.

Reply:
Although, I have just checked and `fs::exists` does resolve symlinks and a quick test of `fs::is_directory` shows that it does also. Should I remove the symlink-resolving code?

Reply:
If fs:exists officially checks, then yes. It makes sense to reply on official documentation. In this case, see https://en.cppreference.com/w/cpp/filesystem/exists, corresponding behaviour is documented. For is_directory it is not as obvious: https://en.cppreference.com/w/cpp/filesystem/is_directory

Reply:
cppreference is a great resource, and I have checked that the C++ standard agrees just to be sure: `is_directory` should be implemented in terms of `fs::status` and that should resolve symlinks.

Should I remove symlink resolving code then? Maybe put a comment with a link to cppreference, to make things clear?

Reply:
If your code correctly handles symlinks without explicit check, then yes, you can remove the symlink specific code. And yes, a comment in this case IMHO is helpful, to make sure, that we do not get an "improvement" pull request wtihout need.

Reply:
We

Reply:
>If your code correctly handles symlinks without explicit check, then yes, you can remove the symlink specific code

According to `[fs.op.exists]` in the standard, `std::fs::exists` is implemented in terms of `std::fs::status`:
```
bool exists(const path& p, error_code& ec) noexcept;
Let s be a file_status, determined as if by status(p) or status(p, ec), respectively.
Effects: The signature with argument ec calls ec.clear() if status_known(s).
Returns: exists(s).
```
The same for `std::fs::is_directory` in `[fs.op.exists]`:
```
bool is_directory(const path& p);
bool is_directory(const path& p, error_code& ec) noexcept;
Returns: is_directory(status(p)) or is_directory(status(p, ec)), respectively. The signature
with argument ec returns false if an error occurs.
```
And according to `[fs.op.status]`, `std::fs::status` resolves symlinks:
```
file_status status(const path& p, error_code& ec) noexcept;

...

Remarks: If a symbolic link is encountered during pathname resolution, pathname resolution continues
using the contents of the symbolic link.
```
So, as long as compiler implementers adhere to the standard, I'm convinced, that both those functions resolve symlinks. I will remove the symlink-resolving code. It was completely wrong anyway - I have created the `resolvedSymlink` variable, which I then failed to use. It seems that it went through the CI linters just fine though...

> And yes, a comment in this case IMHO is helpful, to make sure, that we do not get an "improvement" pull request wtihout need.

Alright, I'll add a comment with a short explanation and a link to cppreference.

Reply:
I have also changed the error message to indicate that `lookupDir` might not be a directory.

Reply:
Sorry, I was no looking to documentation before this comment. Thank you to do so :-)